src/tdastro/astro_utils/snia_utils.py

```diff
@@ -0,0 +1,195 @@
+from astropy.cosmology import FlatLambdaCDM
+from tdastro.base_models import FunctionNode
+import numpy as np
+from scipy.stats.sampling import NumericalInversePolynomial
+from scipy.stats import norm
+
+class HostmassX1Distr:
+    def __init__(self, hostmass):
+        self.hostmass = hostmass
+
+    def _p(self,x1,hostmass = 9.):
```

Comment:
You will need to run the linter to fix some of the formatting.

If you don't have ruff installed, you can use `pip install ruff`

Then just run:
```
ruff format
ruff check --fix
```
and it will autocorrect most of the formatting.